experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,199 @@
+	flag.StringVar(&o.repo, "repo", "", "GitHub repository")
+	flag.Parse()
+	return o
+}
+
+func sanityCheckFlags(o options) error {
+	if o.num <= 0 {
+		return fmt.Errorf("empty or invalid --num")
+	}
+	if o.org == "" {
+		return fmt.Errorf("empty --org")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --repo")
+	}
+	if o.githubTokenFile == "" {
+		return fmt.Errorf("empty --github-token-file")
+	}
+	if o.jobName == "" {
+		return fmt.Errorf("empty --job-name")
+	}
+
+	if o.jenkinsBearerTokenFile == "" && (o.jenkinsUserName == "" || o.jenkinsTokenFile == "") {
+		return fmt.Errorf("neither --jenkins-bearer-token-file nor the combination of --jenkins-user-name and --jenkins-token-file were provided")
+	}
+
+	if o.githubEndpoint == "" {
+		return fmt.Errorf("empty --github-endpoint")
+	} else if _, err := url.Parse(o.githubEndpoint); err != nil {
+		return fmt.Errorf("bad --github-endpoint provided: %v", err)
+	}
+
+	if o.jenkinsURL == "" {
+		return fmt.Errorf("empty --jenkins-url")
+	} else if _, err := url.Parse(o.jenkinsURL); err != nil {
+		return fmt.Errorf("bad --jenkins-url provided: %v", err)
+	}
+
+	return nil
+}
+
+func loadToken(path string) (string, error) {
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(raw)), nil
+}
+
+func main() {
+	o := flagOptions()
+	err := sanityCheckFlags(o)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// TODO(kargakis): dry this out
+	ac := jenkins.AuthConfig{}
+	if o.jenkinsTokenFile != "" {
+		token, err := loadToken(o.jenkinsTokenFile)
+		if err != nil {
+			log.Fatalf("cannot read file specified by --jenkins-token-file: %v", err)
+		}
+		ac.Basic = &jenkins.BasicAuthConfig{
+			User:  o.jenkinsUserName,
+			Token: token,
+		}
+	} else if o.jenkinsBearerTokenFile != "" {
+		token, err := loadToken(o.jenkinsBearerTokenFile)
+		if err != nil {
+			log.Fatalf("cannot read file specified by --jenkins-bearer-token-file: %v", err)
+		}
+		ac.BearerToken = &jenkins.BearerTokenAuthConfig{
+			Token: token,
+		}
+	} else {
+		log.Fatalf("no jenkins auth token provided")
+	}
+
+	jc := jenkins.NewClient(o.jenkinsURL, &ac)
+
+	token, err := loadToken(o.githubTokenFile)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --ghtoken: %v", err)
+	}
+
+	gc := github.NewClient(token, o.githubEndpoint)
+
+	pr, err := gc.GetPullRequest(o.org, o.repo, o.num)
+	if err != nil {
+		log.Fatalf("Unable to get information on pull request %s/%s#%d: %v", o.org, o.repo, o.num, err)
+	}
+
+	spec := kube.ProwJobSpec{
+		Type: kube.PresubmitJob,
+		Job:  o.jobName,
+		Refs: kube.Refs{
+			Org:     o.org,
+			Repo:    o.repo,
+			BaseRef: pr.Base.Ref,
+			BaseSHA: pr.Base.SHA,
+			Pulls: []kube.Pull{
+				{
+					Number: pr.Number,
+					Author: pr.User.Login,
+					SHA:    pr.Head.SHA,
+				},
+			},
+		},
+
+		Report:         false,
+		Context:        "",
+		RerunCommand:   "",
+		MaxConcurrency: 1,
+	}
+
+	if false {
```

Comment:
Please delete this conditional since it is never entered.

Reply:
Woops >.<